barretenberg/cpp/src/barretenberg/vm2/simulation/execution.hpp

```diff
@@ -80,6 +82,15 @@ class Execution : public ExecutionInterface {
     const std::vector<TaggedValue>& get_inputs() const { return inputs; }
     const TaggedValue& get_output() const { return output; }
 
+    void init_gas_tracker(ContextInterface& context) { gas_tracker = execution_components.make_gas_tracker(context); }
+    GasTrackerInterface& get_gas_tracker() { return *gas_tracker; }
+    GasEvent finish_gas_tracker()
+    {
+        GasEvent event = gas_tracker->finish();
+        gas_tracker = nullptr;
+        return event;
+    }
```

Comment:
please move these bodies to the cpp

and add `assert(... == nullptr);`  (o `!=`) to the start of the functions.

Reply:
also, these can be private

Reply:
while you are at it move inputs outs as well to private please